tests/Util/PhpServicesCreatorTest.php

```diff
@@ -0,0 +1,140 @@
+
+        $phpContainerBuilder = new ContainerBuilder();
+        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $phpLoader->load('services_imports.php');
+
+        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+        $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+        $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+        $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+    }
+
+    public function testLoadResources()
+    {
+        $yamlContainerBuilder = new ContainerBuilder();
+        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $yamlLoader->load('load_ressources.yaml');
+
+        $phpContainerBuilder = new ContainerBuilder();
+        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/source/fixtures_load'));
+        $phpLoader->load('load_ressources.php');
+
+        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+        $this->assertTrue($yamlContainerBuilder->getDefinitions() == $phpContainerBuilder->getDefinitions());
+        $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+        $this->assertTrue($yamlContainerBuilder->getServiceIds() == $phpContainerBuilder->getServiceIds());
+    }
+
+    public function testCasesWhenContainerIsBuiltDifferently()
+    {
+        $source = file_get_contents(__DIR__.'/fixtures/YamlConvert/container_built_differently/reference.yaml');
+        $expectedSource = file_get_contents(__DIR__.'/fixtures/YamlConvert/container_built_differently/reference.php');
+
+        $creator = new PhpServicesCreator();
+
+        $this->assertSame($expectedSource, $creator->convert($source));
+
+        $yamlContainerBuilder = new ContainerBuilder();
+        $yamlLoader = new YamlFileLoader($yamlContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/container_built_differently'));
+        $yamlLoader->load('reference.yaml');
+
+        $phpContainerBuilder = new ContainerBuilder();
+        $phpLoader = new PhpFileLoader($phpContainerBuilder, new FileLocator(__DIR__.'/fixtures/YamlConvert/container_built_differently'));
+        $phpLoader->load('reference.php');
+
+        $this->assertTrue($yamlContainerBuilder->getParameterBag() == $phpContainerBuilder->getParameterBag());
+        $this->assertTrue($yamlContainerBuilder->getAliases() == $phpContainerBuilder->getAliases());
+        $this->assertTrue($yamlContainerBuilder->getAutoconfiguredInstanceof() == $phpContainerBuilder->getAutoconfiguredInstanceof());
+    }
```

Comment:
I would also like to just allow the main test method to handle this case. However, I realize that with this test, the container can't be compared in the same way. To handle that, in `getConfigurationFiles()`, I would add a special case there: I would check to see if the filename is `reference.yaml` and, if it is, yield a 3rd `false` argument that will be passed to the test method. Something like:

```php
yield $nameTestCase => [
    file_get_contents($file->getPath().'/source/'.$nameTestCase.'.yaml'),
    file_get_contents($file->getPath().'/'.$file->getFilename()),
    $file->getFilename() === 'reference.yaml' ? false : true,
];
```

The 3rd argument to the test method would now be `bool $shouldCompareDefinitions`. And you can use that to NOT compare the definitions when this flag is false.

Reply:
As suggested in slack, imports and resources are all loaded in a new method testFixturesLoad(). I created a method for comparing containers so this is, i think, short and clear. look if you like.